### DISCUSSION TITLE
Controlled burst load test

### DISCUSSION THREAD

**Discussion by miliev1012** [locustio/locust#2724] [Q&A]
I'm trying to replicate a behavior of a massive spike of load (let say 200K events/s) via a message provider that should last totally 2-3-4 seconds and then go back to lower/normal levels(2-10K). The problem is when I start that test there is a delay(due to initial creation of Users) and then I may eventually reach those levels, but not start directly with that load.
I start wandering is there a way to actually create my test to wait some period(or when ALL users are available) and then blast the load ?
I'm using k8s + master/slave, so maybe going with A LOT of workers is also an option, but I prefer to have a way to provide stable RPS for that initial blast of load. Also those numbers are current and hopefully(for the business) will need to test some times more than that :).

**Comment by cyberw** [comment]
Hi @miliev1012 !

You can delay users actually starting their work using a semaphore, as shown in this example: https://github.com/locustio/locust/blob/master/examples/semaphore_wait.py

The spawning_complete event was kind of broken in distributed mode until just now, so you need the very latest and greatest locust (`pip install --pre -U locust`).

I dont have a way to reduce the load after that but it shouldnt be too hard. Perhaps record the timestamp when the spawning_complete event was received and run `time.sleep(x)` in the task if current time is 5 more than that.
